server/handler/base.go

```diff
@@ -65,19 +73,45 @@ func (p *PullEvaluationOptions) FillDefaults() {
 	}
 }
 
-func (b *Base) MakeStatus(state, message string, pr *github.PullRequest) *github.RepoStatus {
-	var details string
+func (b *Base) PostStatus(ctx context.Context, client *github.Client, owner, repo, ref string, state, message string, pr *github.PullRequest) error {
+	logger := zerolog.Ctx(ctx)
+
+	var detailsURL string
 	if pr != nil {
 		publicURL := strings.TrimSuffix(b.BaseConfig.PublicURL, "/")
-		details = fmt.Sprintf("%s/details/%s/%d", publicURL, pr.GetBase().GetRepo().GetFullName(), pr.GetNumber())
+		detailsURL = fmt.Sprintf("%s/details/%s/%d", publicURL, pr.GetBase().GetRepo().GetFullName(), pr.GetNumber())
+	}
+
+	contextWithBranch := fmt.Sprintf("%s: %s", b.PullOpts.StatusCheckContext, pr.GetBase().GetLabel())
+	status := &github.RepoStatus{
+		Context:     &contextWithBranch,
+		State:       &state,
+		Description: &message,
+		TargetURL:   &detailsURL,
+	}
+
+	logger.Info().Msgf("Setting status context=%s state=%s description=%s target_url=%s", contextWithBranch, state, message, detailsURL)
+	if _, _, err := client.Repositories.CreateStatus(ctx, owner, repo, ref, status); err != nil {
+		return err
 	}
 
-	return &github.RepoStatus{
+	if !b.PullOpts.StatusCheckLegacyContext {
+		return nil
+	}
+
+	legacyStatus := &github.RepoStatus{
 		Context:     &b.PullOpts.StatusCheckContext,
 		State:       &state,
 		Description: &message,
-		TargetURL:   &details,
+		TargetURL:   &detailsURL,
+	}
+
+	logger.Info().Msgf("Setting status context=%s state=%s description=%s target_url=%s", b.PullOpts.StatusCheckContext, state, message, detailsURL)
+	if _, _, err := client.Repositories.CreateStatus(ctx, owner, repo, ref, legacyStatus); err != nil {
```

Comment:
Is it worth defining a private method that creates the `RepoStatus`, logs this message, and makes the API call to avoid duplication?

Reply:
yea, going to clean this up a bit more, which should make the flow less exit-y